src/main/java/athena/TimeAllocator.java

```diff
@@ -23,35 +25,151 @@
+                ArrayList<Integer> temp = new ArrayList<Integer>(Collections.nCopies(space, -1));
+                ArrayList<Task> tasksToRemove = new ArrayList<Task>();
+                while (temp.contains(-1)) {
+                    for (Task currTask : undefinedTimeTasks) {
+                        Integer span = currTask.getTimeInfo().getDuration();
+                        if (span <= space) {
+                            int taskNumber = currTask.getNumber();
+                            this.flexibleTaskList.getTaskFromNumber(taskNumber)
+                                    .getTimeInfo().setStartTime(LocalTime.of(end - space, 0));
+                            for (int i = 0; i < span; i++) {
+                                temp.set(i + end - space - pos, currTask.getNumber());
+                            }
+                            space -= span;
+                            tasksToRemove.add(currTask);
+                            if (space == 0) {
+                                break;
+                            }
+                        }
+                    }
+                    removeAssignedTasks(undefinedTimeTasks, tasksToRemove);
+                    setTempToPerm(dayLog, temp);
+                    if (start == end) {
+                        break;
+                    } else {
+                        start = end;
+                    }
+                    if (undefinedTimeTasks.size() == 0) {
+                        done = true;
+                        break;
+                    }
+                }
+
+            }
+            //System.out.println(fixedDayTasks.getTasks());
+            //System.out.println(flexibleDayTasks.getTasks());
+            currDay.plusDays(1);
+        }
+
+    }
+
+    private ArrayList<Task> getSortedFixedTasks(TaskList taskList) {
+        TaskList fixedDayTasks = taskList;
+        ArrayList<Task> sortedTimeTasks = fixedDayTasks.getTasks();
+        sortedTimeTasks.sort(new TaskTimeComparator());
+        sortedTimeTasks.sort(new TaskImportanceComparator());
+        return sortedTimeTasks;
+    }
+
+    private void removeAssignedTasks(ArrayList<Task> undefinedTimeTasks, ArrayList<Task> tasksToRemove) {
+        for (Task remTask : tasksToRemove) {
+            undefinedTimeTasks.remove(remTask);
+        }
+    }
+
+    private void setTempToPerm(ArrayList<Integer> dayLog, ArrayList<Integer> temp) {
+        int count = 0;
+        for (int taskNumber : temp) {
+            dayLog.set(count, taskNumber);
+            count++;
+        }
+    }
+
+
+    private TaskList getFixedDayTasks(LocalDate date) {
+        ForecastFilter forecast = new ForecastFilter(Forecast.DAY);
+        forecast.setDate(date);
+        TaskList a = this.fixedTaskList.getFilteredList(forecast);
+        return a;
+    }
 
+    private TaskList getFlexibleDayTasks(LocalDate date) {
+        ForecastFilter forecast = new ForecastFilter(Forecast.DAY);
+        forecast.setDate(date);
```

Comment:
I realized that when the `Forecast` is `DAY` then it must have a date. So, it would be good to have it part of the constructor. How about having a second constructor that takes in a `LocalDate`, which sets the forecast to be `Forecast.DAY`.